etnawrapper/etna.py

```diff
@@ -165,9 +165,11 @@ def get_picture(self, login: str = None) -> BytesIO:
         result = self._query(url, raw=True)
         return result.content
 
-    def get_projects(self, login: str = None) -> dict:
+    def get_projects(self, login: str = None, date: datetime = None) -> dict:
         """Fetch a student's projects base on the login."""
         url = SEARCH_URL.format(login=login or self.login)
+        if date is not None:
+            url += "?date=" + date
         result = self._query(url)
```

Comment:
Tu peux utiliser l'argument `params` de `Etnawrapper._query` :)

```suggestion
        params = dict()
        if date is not None:
            _date = date.strftime('%Y-%m-%d')
            params["date"] = _date
        result = self._query(url, params=params)
```

Je te code ca de tete, normalement ca traduit mieux ce a quoi je pensais.

Implicitement, si je dev et que j'ai un argument `date`, je m'attend soit a:
- pouvoir passer un type `datetime`
- pouvoir passer une string de date RFC8601-compliant

Si ca te convient comme approche, c'est parti :)

Reply:
Ok je comprends mieux, tout est ok !